### PR TITLE
chore: Use bun compile for the api server #974

### PR DESCRIPTION
**Pull request by Ratnadeep2k**
Use Bun to compile TypeScript file

### CODE REVIEW

File: apps/server/Dockerfile [hunk 1]
```diff
@@ -15,8 +15,11 @@ RUN rm -rf /app/packages/integrations/vercel
 #  Temp solution for bun
 RUN bun install --ignore-scripts
 
+# Compile the TypeScript application
+RUN bun build --compile --sourcemap src/index.ts --outfile=app
```
Comment (thibaultleouay) [inline apps/server/Dockerfile:19]: we should run this command in the `apps/server` directory 
there is no index.ts found

Comment (Ratnadeep2k) [reply apps/server/Dockerfile:19]: Need to replace src/index.ts to apps/server !

File: apps/server/Dockerfile [hunk 2]
```diff
@@ -15,8 +15,11 @@ RUN rm -rf /app/packages/integrations/vercel
 #  Temp solution for bun
 RUN bun install --ignore-scripts
 
+# Compile the TypeScript application
+RUN bun build --compile --sourcemap apps/server --outfile=app
```
Comment (thibaultleouay) [inline apps/server/Dockerfile]: You need to give the actual path of the file 
you should do --sourcemap src/index.ts but in the directory app server 

you should move this line to line 24

Comment (Ratnadeep2k) [reply apps/server/Dockerfile]: > check once if it works 
@thibaultleouay

File: apps/web/src/components/content/pre.tsx [hunk 3]
```diff
@@ -31,7 +30,7 @@ export default function Pre({ children, ...props }: PreProps) {
   };
 
   return (
-    <div className="relative overflow-hidden">
+    <div className="relative hidden overflow-hidden dark:[&:has([data-theme='dark'])]:block [&:has([data-theme='light'])]:block dark:[&:has([data-theme='light'])]:hidden">
```
Comment (thibaultleouay) [inline apps/web/src/components/content/pre.tsx]: Can you remove this unrelated change please

File: apps/web/src/contentlayer/plugins/pretty-code.ts [hunk 4]
```diff
@@ -7,7 +7,10 @@ import type * as unified from "unified";
 const prettyCode: unified.Pluggable<any[]> = [
   rehypePrettyCode,
   {
-    theme: "github-light",
+    theme: {
```
Comment (thibaultleouay) [inline apps/web/src/contentlayer/plugins/pretty-code.ts]: same for this one if you can remove it

**Review by thibaultleouay — COMMENTED** [review]
Just remove the 2 unrelated commit and it's 🔥

**Review by thibaultleouay — APPROVED** [review]
thanks 🙏

**Comment by Ratnadeep2k** [comment]
> thanks 🙏

Thank you for your helping hand ! Appreciated
